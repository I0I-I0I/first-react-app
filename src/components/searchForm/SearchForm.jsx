import InputWithLabel from "../ui/inputWithLabel/InputWithLabel";
import Button from "../ui/button/Button";
import styles from "./SearchForm.module.css";

const SearchForm = ({ onSearchSubmit, searchTerm, onSearchInput }) => (
	<form onSubmit={onSearchSubmit} className={styles.searchForm}>
		<InputWithLabel
			id="search"
			value={searchTerm}
			onInputChange={onSearchInput}
			isFocused
		>
			Search
		</InputWithLabel>

		<Button type="submit" size="large" disabled={!searchTerm}>
			Search
		</Button>
	</form>
);

export default SearchForm;
