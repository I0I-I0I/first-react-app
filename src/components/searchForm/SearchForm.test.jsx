import { fireEvent, render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import SearchForm from "./SearchForm";

describe("SearchForm", () => {
	const searchFormProps = {
		searchTerm: "React",
		onSearchInput: vi.fn(),
		onSearchSubmit: vi.fn(),
	};

	it("renders the input field with its value", () => {
		render(<SearchForm {...searchFormProps} />);
		expect(screen.getByDisplayValue("React")).toBeInTheDocument();
		// screen.debug();
	});

	it("renders the correct label", () => {
		render(<SearchForm {...searchFormProps} />);
		expect(screen.getByLabelText(/Search/)).toBeInTheDocument();
	});

	it("calls onSearchInput on input field change", () => {
		render(<SearchForm {...searchFormProps} />);
		fireEvent.change(screen.getByDisplayValue("React"), {
			target: { value: "Redux" },
		});
		expect(searchFormProps.onSearchInput).toHaveBeenCalledTimes(1);
	});

	it("calls onSearchSubmit on button submit click", () => {
		render(<SearchForm {...searchFormProps} />);
		fireEvent.submit(screen.getByRole("button"));
		expect(searchFormProps.onSearchSubmit).toHaveBeenCalledTimes(1);
	});

	it("renders snapshot", () => {
		const { container } = render(<SearchForm {...searchFormProps} />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
