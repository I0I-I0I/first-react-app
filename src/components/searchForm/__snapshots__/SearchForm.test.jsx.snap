// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchForm > renders snapshot 1`] = `
<form
  class="_searchForm_a551d0"
>
  <label
    class="_label_1ba982"
    for="search"
  >
    Search
  </label>
  Â 
  <input
    class="_input_1ba982"
    id="search"
    placeholder="Enter something..."
    type="text"
    value="React"
  />
  <button
    class="_button_3c5126 _buttonLarge_3c5126"
    type="submit"
  >
    Search
  </button>
</form>
`;
